File: src/main/kotlin/com/komdatis/model/Building.kt
package com.komdatis.model

import jakarta.persistence.*
import jakarta.validation.constraints.NotNull

@Entity
@Table(name = "buildings")
data class Building(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @NotNull
    @Column(name = "first_name")
    var firstName: String,

    @NotNull
    @Column(name = "last_name")
    var lastName: String,

    @NotNull
    @Column(name = "address")
    var address: String,

    @NotNull
    @Column(name = "living_space")
    var livingSpace: Float,

    @OneToMany(mappedBy = "building", cascade = [CascadeType.ALL], orphanRemoval = true)
    var warmth: List<BuildingWarmth> = emptyList(),

    @OneToMany(mappedBy = "building", cascade = [CascadeType.ALL], orphanRemoval = true)
    var warmWater: List<BuildingWarmWater> = emptyList(),

    @NotNull
    @Column(name = "heated_basement")
    var heatedBasement: Boolean,

    @NotNull
    @Column(name = "apartments")
    var apartments: Short
)

File: src/model/building.ts
import mongoose, { Document, Model, Schema, Types } from "mongoose";
import { BuildingWarmWater } from "./building-warm-water";
import { BuildingWarmth } from "./building-warmth";

export interface Building extends Document {
  firstName: string;
  lastName: string;
  address: string;
  livingSpace: number;
  warmth: Types.Array<Document & BuildingWarmth>;
  warmWater: Types.Array<Document & BuildingWarmWater>;
  heatedBasement: boolean;
  apartments: number;
}

const BuildingSchema: Schema = new Schema({
  firstName: { type: String, required: true },
  lastName: { type: String, required: true },
  address: { type: String, required: true },
  livingSpace: { type: Number, required: true },
  warmth: [{ type: Types.ObjectId, ref: "BuildingWarmth" }],
  warmWater: [{ type: Types.ObjectId, ref: "BuildingWarmWater" }],
  heatedBasement: { type: Boolean, required: true },
  apartments: { type: Number, required: true },
});

const BuildingModel: Model<Building> = mongoose.model<Building>(
  "buildings",
  BuildingSchema
);

export default BuildingModel;
